tap_dynamodb/dynamodb.py

```diff
@@ -66,16 +68,40 @@ def setup_aws_client(config):
     boto3.setup_default_session(botocore_session=refreshable_session)
 
 def get_client(config):
+    # if request_timeout is other than 0,"0" or "" then use request_timeout
+    request_timeout = config.get('request_timeout')
+    if request_timeout and float(request_timeout):
+        request_timeout = float(request_timeout)
+    else: # If value is 0,"0" or "" then set default to 300 seconds.
+        request_timeout = REQUEST_TIMEOUT
+    # add the request_timeout in both connect_timeout as well as read_timeout
+    timeout_config = Config(connect_timeout=request_timeout, read_timeout=request_timeout)
     if config.get('use_local_dynamo'):
         return boto3.client('dynamodb',
                             endpoint_url='http://localhost:8000',
-                            region_name=config['region_name'])
-    return boto3.client('dynamodb', config['region_name'])
+                            region_name=config['region_name'],
+                            config=timeout_config   # pass the config to add the request_timeout
+                            )
+    return boto3.client('dynamodb', config['region_name'],
+                        config=timeout_config   # pass the config to add the request_timeout
+                        )
 
 def get_stream_client(config):
+    # if request_timeout is other than 0,"0" or "" then use request_timeout
+    request_timeout = config.get('request_timeout')
+    if request_timeout and float(request_timeout):
+        request_timeout = float(request_timeout)
+    else: # If value is 0,"0" or "" then set default to 300 seconds.
+        request_timeout = REQUEST_TIMEOUT
```

Comment:
@namrata270998, Create a separate function to calculate request timeout based on provided config as we are calculating the same thing in multiple functions.

Reply:
done